lua/vim_dadbod_completion/blink.lua

```diff
@@ -0,0 +1,67 @@
+---@type blink.cmp.Source
+local M = {}
+
+function M.new()
+  return setmetatable({}, { __index = M })
+end
+
+function M:get_trigger_characters()
+  return { '"', '`', '[', ']', '.' }
+end
+
+function M:enabled()
+  local filetypes = { "sql", "mysql", "plsql" }
+  return vim.tbl_contains(filetypes, vim.bo.filetype)
+end
+
+function M:get_completions(ctx, callback)
+  local cursor_col = ctx.cursor[2]
+  local line = ctx.line
+  local word_start = cursor_col
+
+  local triggers = self:get_trigger_characters()
+  while word_start > 1 do
+    local char = line:sub(word_start - 1, word_start - 1)
+    if vim.tbl_contains(triggers, char) or char:match('%s') then
+      break
+    end
+    word_start = word_start - 1
+  end
+
+  -- Get text from word start to cursor
+  local input = line:sub(word_start, cursor_col)
+
+  local transformed_callback = function(items)
+    callback({
+      context = ctx,
+      is_incomplete_forward = false,
```

Comment:
I'm not sure what's the difference between forward and backward, but we might need to have this always as `true` to correctly get all the results. Similar thing is done for cmp source https://github.com/kristijanhusak/vim-dadbod-completion/blob/master/lua/vim_dadbod_completion.lua?plain=1#L65

Reply:
I mirrored this callback implementation based on how `blink.cmp` uses the callback for buffer completions [here](https://github.com/Saghen/blink.cmp/blob/main/lua/blink/cmp/sources/buffer.lua#L72-L91) and also how it was implemented within `lazydev.nvim` [here](https://github.com/folke/lazydev.nvim/blob/main/lua/lazydev/integrations/blink.lua#L22-L32).

I confess that I don't know exactly how `blink.cmp` treats the `is_incomplete_forward` and `is_incomplete_backward`. Looking through their codebase, I could only find it used with `true` within the [lsp provider](https://github.com/Saghen/blink.cmp/blob/f00a61a43be7309691bee9c147bd1d8700b848a9/lua/blink/cmp/sources/lsp.lua)

Reply:
I understand the forward part, but I'm not sure what's backward.
To make sure it works, let's put both to `true`. If you run into some performance issues we can revisit, but since cmp source does the same I think we will be ok.